crates/ursa-network/src/behaviour.rs

```diff
@@ -94,13 +101,21 @@ pub struct Behaviour<P: StoreParams> {
 
     /// request/response protocol implementation for [`UrsaProtocol`]
     pub(crate) request_response: RequestResponse<UrsaExchangeCodec>,
+
+    /// Graphsync for efficiently exchanging data between blocks between peers.
+    pub(crate) graphsync: GraphSync<G>,
```

Comment:
@b0xtch This is generic to support `Rocksdb` or `MemoryDB` or other implementer of `GSBlockstore`. Is this what you meant? Let me know if I misunderstood.

Reply:
Yeah, the underlying store is the same. So we can abstract it to that layer and then instantiate. Similar to this https://github.com/fleek-network/ursa/blob/ab3e8e407428f84085496f25d9771cf98b79d3e8/crates/ursa-network/src/service.rs#L250

Reply:
So I understand clearly. Are you suggesting wrapping `store` in an abstraction like `GraphsyncStorage` so we could have it like this in `Behaviour`?
```rust
pub(crate) graphsync: GraphsyncStorage,
```
This would still need us to parameterize and bound because both are parameterized as well`GraphsyncStorage<S: Blockstore>` and `BitswapStorage<S: Blockstore>`.

`libp2p_bitswap::Bitswap` doesn't maintain ownership of `store` so that is why we currently don't bound for `Blockstore`.

Reply:
Or maybe you mean something like ?
```rust
struct Behaviour<S: Blockstore, P> {
...
    pub(crate) graphsync: GraphSync<GraphsyncStorage<S>>
```

Reply:
The latter for sure

Reply:
Updated. A nice upgrade to `rs-graphsync` would be for it to handle `store` like `bitswap` does - without mainting ownership. It would allow us to remove the bound.

Reply:
Sounds good 👌